### DISCUSSION TITLE
Docker 部署本地127.0.0.1(localhost)地址无法使用，但是换成内网的IP地址可以正常访问

### DISCUSSION THREAD

**Discussion by ryanzhaorx** [lobehub/lobehub#1677] [Self Hosting] [answered]
### 💻 Operating System

macOS

### 📦 Environment

Docker

### 🌐 Browser

Chrome

### 🐛 Bug Description

本地127.0.0.1(localhost)地址无法使用，但是换成内网的IP地址可以正常访问
{
  "error": {
    "cause": {
      "errno": -111,
      "code": "ECONNREFUSED",
      "syscall": "connect",
      "address": "127.0.0.1",
      "port": 3000
    }
  },
  "endpoint": "http://127.0.***.1:****/v1",
  "provider": "openai"
}

### 🚦 Expected Behavior

能够正常使用localhost(127.0.0.1)

### 📷 Recurrence Steps

<img width="1053" alt="image" src="https://github.com/lobehub/lobe-chat/assets/103013573/f1d85a93-f3f6-4173-9816-d0886b7ed33f">

### 📝 Additional Information

NONE

[a comment by an automated account is not shown]

**Comment by ryanzhaorx** [comment]
<img width="1061" alt="image" src="https://github.com/lobehub/lobe-chat/assets/103013573/85df5fc8-9feb-4014-a987-8bd879b3befe">

**Comment by mingyue0094** [comment, marked as answer]
你是在 Docker 里面装的 `lobehub` 所以，你这里写 `127.**` 他访问的是他自己（lobehub容器本身），不是宿主机。 这不算是bug，也修不了。

[a comment by an automated account is not shown]

**Comment by ryanzhaorx** [comment]
> 你是在 Docker 里面装的 `lobehub` 所以，你这里写 `127.**` 他访问的是他自己（lobehub容器本身），不是宿主机。 这不算是bug，也修不了。

那我想访问宿主机的4000端口怎么设置？

[a comment by an automated account is not shown]

**Comment by MapleEve** [comment]
Docker之间互通是需要映射容器端口到宿主机端口的，这样Docker访问其他的容器是使用宿主机的内网 `IP:port`
不同宿主机之间也可以用宿主机的映射出端口通信，如果外网访问则需要宿主机外网端口再映射到公网IP端口
Emm……可以去学习下Docker原理

[a comment by an automated account is not shown]
